### PR TITLE
rfc: edit provider configuration to add resource conditionally

### PR DESCRIPTION
**Pull request by singhvikash11**

### CODE REVIEW

File: docs/docs/rfcs/00000000_add_resource_conditionally.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Add a scope of condition in provider configuration to filter out on resources.
+
+#### Using expression
+
+Expression can be used to check whether Guardian should add the resources or not.
+
+```yaml
+resources:
+  add_if: $resource.urn endsWith 'keyset'
```
Comment (rahmatrhd) [inline docs/docs/rfcs/00000000_add_resource_conditionally.md:25]: can we rename this field to `filter`?

**Review by rahmatrhd — CHANGES_REQUESTED** [review]

File: docs/docs/rfcs/00000000_add_resource_conditionally.md (hunk 1 above)
Comment (singhvikash11) [reply docs/docs/rfcs/00000000_add_resource_conditionally.md:25]: @rahmatrhd sure, what you suggest for this filter name

Comment (ravisuhag) [reply docs/docs/rfcs/00000000_add_resource_conditionally.md:25]: @singhvikash11 `filter` itself.

Comment (bsushmith) [reply docs/docs/rfcs/00000000_add_resource_conditionally.md:25]: Wouldn't `add_if` be more appropriate here? `filter` can mean either `filtering out` or `include these only`. `add_if` wouldn't have that confusion.

Comment (mabdh) [reply docs/docs/rfcs/00000000_add_resource_conditionally.md:25]: `filter` is fine I guess and can be utilized for generic purpose, or we can use `where` to make it more concise/

Comment (rahmatrhd) [reply docs/docs/rfcs/00000000_add_resource_conditionally.md:25]: @bsushmith the `filter` field will have expression value which expected to return boolean, from there we can infer whether the resource will be included or filtered out. Same as javascript's `Array.filter()` api

**Comment by rahmatrhd** [comment]
converted this PR to an issue here: https://github.com/odpf/guardian/issues/313

**Pull request closed without merging**